dipy/nn/histo_resdnn.py

```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/python
+"""
+Class and helper functions for fitting the Histological ResDNN model.
+"""
+
+
+from packaging.version import Version
+import logging
+
+from dipy.core.gradients import unique_bvals_magnitude, get_bval_indices
+from dipy.core.sphere import HemiSphere
+from dipy.data import get_sphere, get_fnames
+from dipy.reconst.shm import sf_to_sh, sh_to_sf, sph_harm_ind_list
+from dipy.testing.decorators import doctest_skip_parser
+from dipy.utils.optpkg import optional_package
+import numpy as np
+
+tf, have_tf, _ = optional_package('tensorflow')
+if have_tf:
+    from tensorflow.keras.models import Model
+    from tensorflow.keras.layers import Input, Dense, Add
+    if Version(tf.__version__) < Version('2.0.0'):
+        raise ImportError('Please upgrade to TensorFlow 2+')
+
+
+logging.basicConfig()
+logger = logging.getLogger('histo_resdnn')
+
+
+def set_logger_level(log_level):
+    """ Change the logger of the HistoResDNN to one on the following:
+    DEBUG, INFO, WARNING, CRITICAL, ERROR
+
+    Parameters
+    ----------
+    log_level : str
+        Log level for the HistoResDNN only
+    """
+    logger.setLevel(level=log_level)
+
+
+class HistoResDNN():
+    """
+    This class is intended for the ResDNN Histology Network model.
+    """
+
+    @doctest_skip_parser
+    def __init__(self, sh_order=8, basis_type='tournier07', verbose=False):
+        r"""
+        The model was re-trained for usage with a different basis function
+        ('tournier07') like the proposed model in [1, 2].
+
+        To obtain the pre-trained model, use::
+        >>> resdnn_model = HistoResDNN()
+        >>> fetch_model_weights_path = get_fnames('histo_resdnn_weights')
+        >>> resdnn_model.load_model_weights(fetch_model_weights_path)
```

Comment:
for each line, the following comment is missing: ` # skip if not have_tf `. 

The decorator `doctest_skip_parser` will catch it and ignore it for the test.

Reply:
Thanks, I missed that part!